test/unit/tracks/tracks.test.js

```diff
@@ -342,3 +342,28 @@ if (Html5.supportsNativeTextTracks()) {
     emulatedTt.on('addtrack', addtrack);
   });
 }
+
+test('should check for text track changes when using non native text tracks', function() {
+  let origFeaturesNativeTextTracks = Tech.prototype['featuresNativeTextTracks'];
+  Tech.prototype['featuresNativeTextTracks'] = true;
+  let Player = Component.getComponent('Player');
+  let origTextTracks = Player.prototype.textTracks;
+  let numTextTracksCalls = 0;
+  Player.prototype.textTracks = function() {
```

Comment:
How about listening for `texttrackchange` events instead? It seems like then you wouldn't have to instrument the `textTracks` function and could remove the shadowed `tracks` variable.